File: 206-reverse-linked-list/206-reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode current = head; 
        ListNode answer = null;
        
        while(current != null)
        {
            ListNode item = new ListNode(current.val);
            
            item.next = answer; 
            answer = item; 
            
            current = current.next;
        }
        
        return answer;
    }
}

File: 206-reverse-linked-list/206-reverse-linked-list.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ReverseList(ListNode head) {
        ListNode pointer = null; 
        ListNode current = head;
        
        while ( current != null )
        {
            ListNode link = new ListNode(current.val);
            
            link.next = pointer;
            pointer = link; 
            
            current = current.next; 
        }
        
        return pointer; 
    }
}